extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/mutiny/MutinyTracingHelper.java

```diff
@@ -0,0 +1,119 @@
+     * .onItem().transformToUni(s -> wrapWithSpan(tracer, "mySpan", Uni.createFrom().item(s + " World")))
+     * .subscribe().with(System.out::println);
+     * ```
+     * <p>
+     * it also works with multi:
+     * ```java
+     * Multi.createFrom().items("Alice", "Bob", "Charlie")
+     * .onItem().transform(name -> "Hello " + name)
+     * .subscribe().with(System.out::println);
+     * ```
+     * wrap like this:
+     * ```java
+     * Multi.createFrom().items("Alice", "Bob", "Charlie")
+     * .onItem().transformToUni(s -> wrapWithSpan(tracer, "mySpan", Uni.createFrom().item("Hello " + s)
+     * .onItem().transform(name -> "Hello " + name)
+     * ))
+     * .subscribe().with(System.out::println);
+     * ```
+     *
+     * @param <T> the type of the result of the pipeline
+     * @param spanName
+     *        the name of the span that should be created
+     * @param pipeline
+     *        the pipeline to run within the span
+     *
+     * @return the result of the pipeline
+     */
+    public static <T> Uni<T> wrapWithSpan(final Tracer tracer, final String spanName, final Uni<T> pipeline) {
+
+        return wrapWithSpan(tracer, Optional.of(io.opentelemetry.context.Context.current()), spanName, pipeline);
+    }
+
+    /**
+     * see {@link #wrapWithSpan(Tracer, String, Uni)}
+     * use this method if you manually want to specify the parent context of the new span
+     * or if you want to ensure the new span is a root span.
+     *
+     * @param <T>
+     * @param parentContext
+     *        the parent context to use for the new span. If empty, a new root span will be created.
+     * @param spanName
+     *        the name of the span that should be created
+     * @param pipeline
+     *        the pipeline to run within the span
+     *
+     * @return the result of the pipeline
+     */
+    public static <T> Uni<T> wrapWithSpan(final Tracer tracer,
+            final Optional<io.opentelemetry.context.Context> parentContext,
+            final String spanName, final Uni<T> pipeline) {
+
+        return runOnDuplicatedContext(Uni.createFrom().deferred(() -> {
+            final SpanBuilder spanBuilder = tracer.spanBuilder(spanName);
+            if (parentContext.isPresent()) {
+                spanBuilder.setParent(parentContext.get());
+            } else {
+                spanBuilder.setNoParent();
+            }
+            final Span span = spanBuilder.startSpan();
+            final Scope scope = span.makeCurrent();
+            return pipeline.onTermination()
+                    .invoke((o, throwable, isCancelled) -> {
+                        if (Boolean.TRUE.equals(isCancelled)) {
+                            span.recordException(new CancellationException());
+                        } else if (throwable != null) {
+                            span.recordException(throwable);
+                        }
+                        span.end();
+                        scope.close();
```

Comment:
I think the scope close should be on a finally block. It's a good practice.
See https://github.com/quarkusio/quarkus/blob/6ccff484bc930f393a97f61de3391a8b1967ece6/extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/cdi/WithSpanInterceptor.java#L117

Reply:
Nice catch!